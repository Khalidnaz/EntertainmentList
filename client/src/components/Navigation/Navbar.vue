<template>
  <v-app-bar app clipped-left>
    <v-app-bar-nav-icon @click.stop="drawer = !drawer" />
    <v-toolbar-title>Entertainment List</v-toolbar-title>
  </v-app-bar>
</template>

<script>
export default {
  name: "Navbar",
  props: ["toggleSideDrawer"],
  created() {
    console.log("props", this.props);
  }
};
</script>