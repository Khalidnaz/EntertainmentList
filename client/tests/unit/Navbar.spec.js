import Vue from 'vue';
import Vuetify from 'vuetify';
import { mount, createLocalVue } from '@vue/test-utils';
import Navbar from '@/components/Navigation/Navbar.vue';

Vue.use(Vuetify);

const localVue = createLocalVue();

describe('Navbar', () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('should render', () => {
    const wrapper = mount(Navbar, {
      localVue,
      vuetify,
      propsData: {
        user: {},
        toggleDrawer: jest.fn(),
      },
    });
    
    expect(wrapper.isVueInstance()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
