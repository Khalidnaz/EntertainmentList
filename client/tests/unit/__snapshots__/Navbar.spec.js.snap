// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should do something 1`] = `
<header class="v-sheet v-sheet--tile theme--light v-toolbar v-app-bar v-app-bar--clipped v-app-bar--fixed" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
  <div class="v-toolbar__content" style="height: 64px;"><button type="button" class="v-app-bar__nav-icon v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu theme--light"></i></span></button>
    <div class="v-toolbar__title">Entertainment List</div>
    <div class="spacer"></div>
    <!---->
    <div class="v-menu"><button type="button" class="v-btn v-btn--flat v-btn--text theme--dark v-size--default white--text"><span class="v-btn__content">Hello </span></button>
      <!---->
    </div>
  </div>
</header>
`;
